File: 96-不同的二叉搜索树.py
class Solution:
    def numTrees(self, n: int) -> int:

        nums = [1, 1]

        for i in range(n - 1):
            num = 0
            for j in range(i + 2):
                num += nums[j] * nums[i + 1 - j]
            nums.append(num)
    
        return nums[n]

File: 96-不同的二叉搜索树.cpp
class Solution {
public:
    int numTrees(int n) {

        vector<int> nums = {1, 1};
        
        for(int i = 0; i < n - 1; i++){
            int num = 0;
            for(int j = 0; j < i + 2; j++){
                num += nums[j] * nums[i + 1 - j];
            }
            nums.push_back(num);
        }

        return nums[n];
    }
};
